File: Derelict2D-trunk/app/src/main/java/br/odb/derelict2d/ShowGameSplashActivity.java
package br.odb.derelict2d;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.content.Intent;
import android.media.MediaPlayer;
import android.os.Bundle;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.Button;

import br.odb.gamelib.android.AndroidUtils;
import br.odb.gamelib.android.GameView;

public class ShowGameSplashActivity extends Activity implements OnClickListener {

	private MediaPlayer theme;
	private GameView gvSplash;
	private GameView gvLogo;
	private Button playBtn;

	@SuppressLint("NewApi")
	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_show_game_splash);

		playBtn = findViewById(R.id.btnPlay);

		gvSplash = findViewById(R.id.gvSplash);
		gvLogo = findViewById(R.id.gvLogo);

		gvSplash.setVisibility(View.INVISIBLE);
		gvLogo.setVisibility(View.INVISIBLE);

		getWindow().getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_LOW_PROFILE);
	}

	private void startLoading() {

		new Thread(new Runnable() {
			@Override
			public void run() {

				runOnUiThread(new Runnable() {

					@Override
					public void run() {
						initImage();
						gvSplash.postInvalidate();
						gvLogo.postInvalidate();

						gvSplash.setVisibility(View.VISIBLE);
						gvLogo.setVisibility(View.VISIBLE);

						playBtn.setText("Play");
						playBtn.setOnClickListener(ShowGameSplashActivity.this);
					}
				});
			}
		}).start();
	}

	@Override
	public void onWindowFocusChanged(boolean hasFocus) {
		super.onWindowFocusChanged(hasFocus);

		startLoading();

		if (theme == null
				&& ((Derelict2DApplication) getApplication()).mayEnableSound()) {
			theme = MediaPlayer.create(this, R.raw.derelicttheme);
			theme.setLooping(true);
			theme.start();
		}
	}

	private void initImage() {

		Derelict2DApplication app = ((Derelict2DApplication) getApplication());

		AndroidUtils.initImage(gvLogo, "logo", app.getAssetManager());
		AndroidUtils.initImage(gvSplash, "title", app.getAssetManager());
	}

	protected void onActivityResult(int requestCode, int resultCode, Intent data) {
		finish();
	}

	@Override
	protected void onPause() {
		if (theme != null) {
			theme.stop();
		}
		super.onPause();
	}

	@Override
	protected void onDestroy() {
		if (theme != null) {

			theme.stop();
		}
		super.onDestroy();
	}

	@Override
	public void onClick(View v) {

		if (theme != null) {
			theme.stop();
		}

		Intent intent = new Intent(this, RootGameMenuActivity.class);
		this.startActivityForResult(intent, 1);
	}
}

File: Derelict2D-trunk/app/src/main/java/br/odb/derelict2d/ShowGameSplashActivity.kt
package br.odb.derelict2d

import android.app.Activity
import android.content.Intent
import android.media.MediaPlayer
import android.os.Bundle
import android.view.View
import android.widget.CheckBox
import android.widget.CompoundButton
import br.odb.gamelib.android.AndroidUtils
import br.odb.gamelib.android.GameView
import kotlinx.android.synthetic.main.activity_show_game_splash.*

class ShowGameSplashActivity : Activity(), View.OnClickListener, CompoundButton.OnCheckedChangeListener {
    private var theme: MediaPlayer? = null

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_show_game_splash)

        chkSound.setChecked((application as Derelict2DApplication)
                .mayEnableSound())

        chkSound.setOnCheckedChangeListener(this)
        gvSplash.setVisibility(View.INVISIBLE)
        gvLogo.setVisibility(View.INVISIBLE)
        btnExploreStation.setOnClickListener(this)
        btnAbout.setOnClickListener(this)
        btnHowToPlay.setOnClickListener(this)
        window.decorView.systemUiVisibility = View.SYSTEM_UI_FLAG_LOW_PROFILE
    }

    private fun startLoading() {
        Thread(Runnable {
            runOnUiThread {
                initImage()
                gvSplash!!.postInvalidate()
                gvLogo!!.postInvalidate()
                gvSplash!!.visibility = View.VISIBLE
                gvLogo!!.visibility = View.VISIBLE
            }
        }).start()
    }

    override fun onWindowFocusChanged(hasFocus: Boolean) {
        super.onWindowFocusChanged(hasFocus)
        startLoading()
        startTheme()
    }

    private fun startTheme() {
        if (theme == null
                && (application as Derelict2DApplication).mayEnableSound()) {
            theme = MediaPlayer.create(this, R.raw.derelicttheme)

            if ( theme != null ) {
                theme!!.setLooping(true)
                theme!!.start()
            }
        }
    }

    private fun initImage() {
        val app = application as Derelict2DApplication
        AndroidUtils.initImage(gvLogo, "logo", app.assetManager)
        AndroidUtils.initImage(gvSplash, "title", app.assetManager)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent) {
        finish()
    }

    override fun onPause() {
        if (theme != null) {
            theme!!.stop()
        }
        super.onPause()
    }

    override fun onDestroy() {
        if (theme != null) {
            theme!!.stop()
        }
        super.onDestroy()
    }

    override fun onClick(v: View) {
        if (theme != null) {
            theme!!.stop()
        }
        val intent: Intent
        val bundle = Bundle()
        bundle.putString("hasSound", if (chkSound!!.isChecked) "y" else "n")
        when (v.id) {
            R.id.btnExploreStation -> {
                (this.application as Derelict2DApplication).startNewGame()
                intent = Intent(this, ExploreStationActivity::class.java)
                intent.putExtras(bundle)
                startActivity(intent)
            }
            R.id.btnHowToPlay -> {
                intent = Intent(this, ShowHowToPlayActivity::class.java)
                startActivity(intent)
            }
            R.id.btnAbout -> {
                intent = Intent(this, ShowCreditsActivity::class.java)
                startActivity(intent)
            }
        }
    }

    override fun onCheckedChanged(buttonView: CompoundButton, isChecked: Boolean) {
        if (!isChecked) {
            if (theme != null) {
                theme!!.stop()
                theme = null
            }
        } else {
            startTheme()
        }
    }
}